packages/gafl-webapp-service/src/pages/multibuy/change-licence-options/__tests__/route.spec.js

```diff
@@ -0,0 +1,391 @@
+      const mockTransaction = mockTransactionCacheGet.mockImplementationOnce(() => ({ licensee: { firstName: 'John' } }))
+      const getDataRedirectError = new GetDataRedirect(NAME.uri)
+      await expect(getData(mockRequest(mockTransaction))).rejects.toThrow(getDataRedirectError)
+    })
+
+    it('should return a redirect error if date of birth is not included on the licensee', async () => {
+      const mockTransaction = mockTransactionCacheGet.mockImplementationOnce(() => ({ licensee: { firstName: 'John', lastName: 'Smith' } }))
+      const getDataRedirectError = new GetDataRedirect(DATE_OF_BIRTH.uri)
+      await expect(getData(mockRequest(mockTransaction))).rejects.toThrow(getDataRedirectError)
+    })
+
+    it('should return a redirect error if start date is not included on the licensee', async () => {
+      const mockTransaction = mockTransactionCacheGet.mockImplementationOnce(() => ({
+        licensee: { firstName: 'John', lastName: 'Smith', birthDate: '1996-01-01' },
+        licenceStartDate: null
+      }))
+      const getDataRedirectError = new GetDataRedirect(LICENCE_TO_START.uri)
+      await expect(getData(mockRequest(mockTransaction))).rejects.toThrow(getDataRedirectError)
+    })
+
+    it('should return a redirect error if number of rods is not included on the licensee', async () => {
+      const mockTransaction = mockTransactionCacheGet.mockImplementationOnce(() => ({
+        licensee: { firstName: 'John', lastName: 'Smith', birthDate: '1996-01-01' },
+        licenceStartDate: '2025-01-01',
+        licenceType: 'Salmon and sea trout'
+      }))
+      const getDataRedirectError = new GetDataRedirect(LICENCE_TYPE_URI.uri)
+      await expect(getData(mockRequest(mockTransaction))).rejects.toThrow(getDataRedirectError)
+    })
+
+    it('should return a redirect error if licence type is not included on the licensee', async () => {
+      const mockTransaction = mockTransactionCacheGet.mockImplementationOnce(() => ({
+        licensee: { firstName: 'John', lastName: 'Smith', birthDate: '1996-01-01' },
+        licenceStartDate: '2025-01-01',
+        numberOfRods: '3'
+      }))
+      const getDataRedirectError = new GetDataRedirect(LICENCE_TYPE_URI.uri)
+      await expect(getData(mockRequest(mockTransaction))).rejects.toThrow(getDataRedirectError)
+    })
+
+    it('should return a redirect error if licence length is not included on the licensee', async () => {
+      const mockTransaction = mockTransactionCacheGet.mockImplementationOnce(() => ({
+        licensee: { firstName: 'John', lastName: 'Smith', birthDate: '1996-01-01' },
+        licenceStartDate: '2025-01-01',
+        numberOfRods: '3',
+        licenceType: 'Salmon and sea trout'
+      }))
+      const getDataRedirectError = new GetDataRedirect(LICENCE_LENGTH.uri)
+      await expect(getData(mockRequest(mockTransaction))).rejects.toThrow(getDataRedirectError)
+    })
```

Comment:
These are really thorough, but they're very similar and, I think, could be replaced by the following:

```
it.each([
  ['first name', NAME.uri, td => { delete td.licensee.firstName; return td }],  
  ['last name', NAME.uri, td => { delete td.lastName; return td }],  
  ['date of birth', DATE_OF_BIRTH.uri, td => { delete td.licensee.birthDate; return td }], 
  ['start date', LICENCE_TO_START.uri, td => { delete td.licenseStartDate; return td }],
  ['number of rods', LICENCE_TYPE_URI.uri, td => { delete td.numberOfRods; return td; }],
  ['licence type', LICENCE_TYPE_URI.uri, td => { delete td.licenceType; return td }],
  ['licence length', LICENCE_LENGTH.uri, td => td]
])('should return a redirect error if %s is not included', (_p, redirectUri, transform) => {
  const transactionData = {
    licensee: { firstName: 'John', lastName: 'Smith', birthDate: '1996-01-01' },
    licenceStartDate: '2025-01-01',
    numberOfRods: '3',
    licenceType: 'Salmon and sea trout'
  }
  const getDataRedirectError = new GetDataRedirect(redirectUri)
  await expect(getData(mockRequest(transform(transactionData))).rejects.toThrow(getDataRedirectError)
})
```
(syntax here may not be 100% as I typed it here)